site/server/app/com/fortysevendeg/exercises/models/UserModel.scala

```diff
@@ -20,81 +20,99 @@ trait UserStore {
-    def id = column[Option[Long]]("id", O.PrimaryKey, O.AutoInc)
-    def login = column[String]("login")
-    def name = column[String]("name")
-    def github_id = column[String]("github_id")
-    def picture_url = column[String]("picture_url")
-    def github_url = column[String]("github_url")
-    def email = column[String]("email")
-    def * = (id, login, name, github_id, picture_url, github_url, email) <> (User.tupled, User.unapply)
+object UserDoobieStore extends UserStore {
+  def all: ConnectionIO[List[User]] = {
+    Queries.all.list
   }
 
-  val users = TableQuery[Users]
-
-  override def all: Future[Seq[User]] = {
-    db.run(users.sortBy(_.id.desc).result)
+  def getByLogin(login: String): ConnectionIO[Option[User]] = {
+    Queries.byLogin(login).option
   }
 
-  override def getByLogin(login: String): Future[Seq[User]] = db.run(users.filter(_.login === login).result)
+  def getById(id: Int): ConnectionIO[Option[User]] = {
+    Queries.byId(id).option
+  }
 
-  override def create(
+  def create(
     login:       String,
     name:        String,
     github_id:   String,
     picture_url: String,
     github_url:  String,
     email:       String
-  ): Future[User] = {
-
-    db.run {
-      (users returning users.map(_.id) into ((user, id) ⇒ user.copy(id = id))) += User(
-        id = None,
-        login = login,
-        name = name,
-        github_id = github_id,
-        picture_url = picture_url,
-        github_url = github_url,
-        email = email
-      )
-    }
+  ): ConnectionIO[Option[User]] = for {
+    _ ← Queries.insert(login, name, github_id, picture_url, github_url, email).run
+    user ← getByLogin(login)
+  } yield user
+
+  def deleteQuery(login: String): ConnectionIO[Boolean] = {
+    Queries.byLogin(login).option.flatMap({
+      case None ⇒ sql"select false".query[Boolean].unique
+      case Some(u) ⇒ Queries.update(
+        u.id.get,
+        u.login,
+        u.name,
+        u.github_id,
+        u.picture_url,
+        u.github_url,
+        u.email
+      ).run.map(_ ⇒ true)
+    })
   }
 
-  override def update(user: User): Future[Boolean] = {
-    db.run {
-      val q = for {
-        u ← users if u.id === user.id
-      } yield (u.login)
-      q.update(user.login)
-    }.map(_ == 1)
-  }
+  def delete(id: Int): ConnectionIO[Boolean] =
+    Queries.delete(id)
+
+  def deleteAll: ConnectionIO[Int] =
+    Queries.deleteAll.run
+
+  def updateQuery(user: User): ConnectionIO[Int] = {
+    require(user.id.isDefined, "Can only update existing users")
 
-  override def delete(ids: Long*): Future[Boolean] = {
-    Future.sequence {
-      for {
-        id ← ids
-      } yield { db.run(users.filter(_.id === id).delete).map(_ == 1) }
-    }.map(!_.exists(i ⇒ !i))
+    Queries.update(
+      user.id.get,
+      user.login,
+      user.name,
+      user.github_id,
+      user.picture_url,
+      user.github_url,
+      user.email
+    ).run
   }
 
+  def update(
+    id:          Int,
```

Comment:
What do you think about having an ADT with two constructors: `NewUser` and `User` instead of making the ID an `Option`? `NewUser` would have all the fields needed to create a user and `User`'s ID wouldn't be an `Option` anymore.